### ISSUE TITLE
Full-text-index using GraphDB Lucene connector

### ISSUE THREAD

**Issue by subotic** [dhlab-basel/Knora#918]
I'm currently working on #898 which raises a new issue. The way how the full-text-index was created up until now does not work anymore. It is not allowed to send system entities (e.g., `luc:include luc:setParam "literal" .`) to a `owlim:MonitorRepository`. The recommended way by Ontotext is to use the `GraphDB Lucene Connector`. Using the connector has also the added benefit, that any changes to a Knora-Resource are automatically synchronized to the Lucene index on a granular level. The implementation before would require a manual triggering of rebuilding the index, which is actually very costly, and I don't know if this would be feasible in the long run anyway (think of a lot of projects with a lot of data and on each update, the whole index is rebuilt).

Now I'm trying to configure the Lucene connector in the GraphDB Workbench. When the connector is configured, then the configuration can be exported and used in our scripts to create the connector when a new repository is created.

As far as I understand it, the connector configuration requires us to more explicit on what to index.

My question is, which literals of top-level classes and properties need indexing? The subclasses and subproperties are also always indexed.

### What to index:

- `knora-base::Resource`
  - `rdfs:label`
- `knora-base:Value`
  - `knora-base:valueHasComment`
- `knora-base:TextValue`
  - `knora-base:valueHasString`

**General restriction**: Only indexed if the type of the literal is `xsd:string`

**Comment by benjamingeer** [comment]
That's what I remember, too. But maybe we shouldn't index the `valueHasString` of all values. Does it really make sense to search for the string representation of a date? I think it would make more sense to index `valueHasString` only if it's in a `TextValue`.

**Comment by tobiasschweizer** [comment]
We should not forget about the comment on values.

**Comment by benjamingeer** [comment]
`knora-base:valueHasComment`

**Comment by subotic** [comment]
I've updated the list.

**Comment by tobiasschweizer** [comment]
> Does it really make sense to search for the string representation of a date?

I think this should be discussed with @lrosenth

**Comment by subotic** [comment]
So, I give up on getting the GraphDB Lucene Connector to work. Also, my initial assumption was that we could use the connector to create the index and then use the index as is in our SPARQL queries. Now I'm not so sure anymore. I think it would require a completely new syntax, which would be a lot of work to do.

Instead, I managed to create the index the same way we did before in conjunction with the correct repository type of `owlim: MonitorRepository`.

So everything should now work as it did before.

There is still the issue with updating the Lucene index #925, which we also had before.
